### PR TITLE
Jormun: Add ridesharing for distributed scenario + tests

### PR DESCRIPTION
**Pull request by mbounabi**
https://jira.kisio.org/browse/NAVP-927

### CODE REVIEW

**Comment by stifoon** [comment]
\o/ !!!
You're THE guy.

File: source/jormungandr/jormungandr/scenarios/tests/new_default_tests.py [hunk 1]
```diff
@@ -501,7 +502,7 @@ def crowfly_in_ridesharing_test():
     section.type = response_pb2.STREET_NETWORK
     section.street_network.mode = response_pb2.Walking
 
-    new_default._switch_back_to_ridesharing(response, True)
+    new_default.switch_back_to_ridesharing(response, True)
```
Comment (xlqian) [inline source/jormungandr/jormungandr/scenarios/tests/new_default_tests.py]: what?

File: source/jormungandr/tests/journey_common_tests.py [hunk 2]
```diff
@@ -1589,3 +1589,60 @@ def test_night_bus_filter_parameters_with_maximum_values_3(self):
...
+        assert 'to_delete' in rs_journey["tags"]  # no response provided for ridesharing: to_delete
+        rs_section = rs_journey["sections"][0]
+        assert rs_section["departure_date_time"] == rs_journey["departure_date_time"]
+        assert rs_section["arrival_date_time"] == rs_journey["arrival_date_time"]
+        assert rs_section["duration"] == rs_journey["duration"]
+        assert rs_section["mode"] == "ridesharing"
+        assert rs_section["type"] == "street_network"
+        assert rs_section["id"] # check that id is provided
+        assert rs_section["geojson"]["properties"][0]["length"] == rs_journey["distances"]["ridesharing"]
+        assert (rs_section["geojson"]["coordinates"][0][0] - float(rs_section["from"]["address"]["coord"]["lon"])) < 0.000001
```
Comment (xlqian) [inline source/jormungandr/tests/journey_common_tests.py]: take a look at this :) 

```python
import pytest

assert 2.2 == pytest.approx(2.3)
```

Comment (xlqian) [reply source/jormungandr/tests/journey_common_tests.py]: i'm not sure that we use `py.test` for integration tests

File: source/jormungandr/tests/journey_common_tests.py [hunk 3]
```diff
@@ -1589,3 +1589,60 @@ def test_night_bus_filter_parameters_with_maximum_values_3(self):
...
+
+        query = "isochrones?from=0.0000898312;0.0000898312&datetime=20120614T075500&"\
+                "first_section_mode[]={first}&last_section_mode[]={last}&max_duration=2"\
+                .format(first='ridesharing', last='walking')
+        exec_and_check(query)
+
+        query = "heat_maps?from=0.0000898312;0.0000898312&datetime=20120614T075500&"\
+                "first_section_mode[]={first}&last_section_mode[]={last}&max_duration=2"\
+                .format(first='ridesharing', last='walking')
+        exec_and_check(query)
```
Comment (xlqian) [inline source/jormungandr/tests/journey_common_tests.py]: newline

File: source/jormungandr/tests/journey_common_tests.py [hunk 4]
```diff
@@ -34,7 +34,8 @@
 from jormungandr import i_manager
 from .check_utils import *
 import mock
-import  os
+import os
```
Comment (xlqian) [inline source/jormungandr/tests/journey_common_tests.py]: module is not used

Comment (mbounabi) [reply source/jormungandr/tests/journey_common_tests.py]: yes it is : 
https://github.com/CanalTP/navitia/pull/2467/files/91d33040c7d35e4a512683688f785f24818272a0#diff-0832703183fb3233e60448ac96ec726bR931

File: source/jormungandr/jormungandr/scenarios/tests/new_default_tests.py (hunk 1 above)
Comment (mbounabi) [reply source/jormungandr/jormungandr/scenarios/tests/new_default_tests.py]: Bad 'Find and replace'...

File: source/jormungandr/tests/journey_common_tests.py (hunk 4 above)
Comment (xlqian) [reply source/jormungandr/tests/journey_common_tests.py]: ok!

**Review by xlqian — APPROVED** [review]

**Review by kadhikari — APPROVED** [review]

**Review by benoit-bst — APPROVED** [review]

**Review by nonifier — APPROVED** [review]

**Pull request merged**